contracts/src/LightClient.sol

```diff
@@ -341,4 +367,77 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice updates the delay threshold which determines if hotshot is down
+    /// @dev if the number of blocks between the last l1 update and the number of blocks that has
+    /// passed since is > the delayThreshold then hotshot is considered down
+    function updateDelayThreshold(uint32 newThreshold) public onlyOwner {
+        if (newThreshold == delayThreshold) {
+            revert NoChangeRequired();
+        }
+        if (newThreshold < 1 || newThreshold > type(uint32).max) {
+            revert InvalidDelayThreshold();
+        }
+        delayThreshold = newThreshold;
+        emit NewDelayThreshold(newThreshold);
+    }
+
+    /// @notice checks the L1BlockUpdates array and the delayThreshold to determine if hotshot was
+    /// down at a specified L1 block number
+    function isHotshotAlive(uint256 l1BlockNumber) public view returns (bool) {
+        // The block being requested is in the future so return false
+        // if(l1BlockNumber>block.number){
+        //     return true;
+        // }
+
+        // If there is only one block recorded then use the block ledger's current block as the
+        // 'nextBlock'
+        if (l1BlockUpdates.length == 1) {
+            if (
+                block.number - l1BlockUpdates[0] >= delayThreshold
```

Comment:
Why are both these checks necessary? Isn't it sufficient to check that `l1BlockNumber - l1BlockUpdates[0] >= delayThreshold`

Reply:
resolving as the implementation has since changed